### PR TITLE
chore(Go): add selection sort

### PR DESCRIPTION
**Pull request by ayo-ajayi**
**PR Checklist:**
- [x] My submission is formatted according to the guidelines in the [contributing guide](https://github.com/MakeContributions/DSA#contribution-guidelines)
- [x] My addition is on refer on the language README.md file
- [x] My addition does not have a spelling problem
- [x] My submission has a proper and user-friendly description of the algorithm
- [x] My submission has the time complexity of the algorithm
- [x] My submission has sample input-output of the program (NOT FOR PYTHON)

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] New algorithm
- [ ] Optimization in previous algorithms
- [ ] Code style update
- [x] Refactor
- [x] Documentation
- [ ] Other, please describe:

**Briefly describe the changes in this PR**

**Other information:**

### CODE REVIEW

File: algorithms/Go/README.md [hunk 1]
```diff
@@ -20,6 +20,7 @@
 - [Bubble Sort](sorting/bubble-sort.go)
 - [Insertion Sort](sorting/insertion-sort.go)
 - [Quicksort](sorting/quicksort.go)
+- [Selectionsort](sorting/selection-sort.go)
```
Comment (ming-tsai) [inline algorithms/Go/README.md:23]: Please separate the word
```suggestion
- [Selection Sort](sorting/selection-sort.go)
```

**Review by ming-tsai — CHANGES_REQUESTED** [review]
Hi @ayo-ajayi,
We don't have `Go` language reviewer, so I won't check your code, but I have a suggestion for the `README.md` file.

File: algorithms/Go/README.md (hunk 1 above)
Comment (ayo-ajayi) [reply algorithms/Go/README.md]: done

**Review by ming-tsai — APPROVED** [review]
@ayo-ajayi thanks, if you want to be a Go language reviewer, please write to me on the #158

**Pull request merged**